### DISCUSSION TITLE
what is compute_sharpness used for in nerf loader?

### DISCUSSION THREAD

**Discussion by neilgogogo** [NVlabs/instant-ngp#154] [Q&A] [answered]
Thank you very much for your work. It's very impressive.

 I have a question about sharpness. what is this function compute_sharpness used for? Can I filter this function?

https://github.com/NVlabs/instant-ngp/blob/09f359765fd8f49fb4ad0dbc1ae2732c00201b8b/src/nerf_loader.cu#L598

Kind regards

**Comment by Tom94** [comment, marked as answer]
Thanks for the kind words! :)

That function is somewhat of an edge detector. We've used it in experiments where we tried importance sampling the NeRF optimization proportional to the image-space sharpness, error, and various combinations thereof -- you can toggle these modes in the GUI.
